src/Microsoft.AspNetCore.OData/Extensions/ODataApplicationBuilderExtensions.cs

```diff
@@ -26,5 +37,50 @@ public static IApplicationBuilder UseODataBatching(this IApplicationBuilder app)
 
             return app.UseMiddleware<ODataBatchMiddleware>();
         }
+
+        /// <summary>
+        /// Use OData route with default route name and route prefix.
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder "/> to use.</param>
+        /// <param name="model">The <see cref="IEdmModel"/> to use.</param>
+        /// <returns>The <see cref="IApplicationBuilder "/>.</returns>
+        public static IApplicationBuilder UseOData(this IApplicationBuilder app, IEdmModel model)
+        {
+            return app.UseOData(DefaultRouteName, DefaultRoutePrefix, model);
+        }
+
+        /// <summary>
+        /// Use OData route with given route name and route prefix.
+        /// </summary>
+        /// <param name="app">The <see cref="IApplicationBuilder "/> to use.</param>
+        /// <param name="routeName">The given OData route name.</param>
+        /// <param name="routePrerix">The given OData route prefix.</param>
+        /// <param name="model">The <see cref="IEdmModel"/> to use.</param>
+        /// <returns>The <see cref="IApplicationBuilder "/>.</returns>
+        public static IApplicationBuilder UseOData(this IApplicationBuilder app, string routeName, string routePrerix, IEdmModel model)
+        {
+            if (app == null)
+            {
+                throw Error.ArgumentNull("app");
+            }
+
+            VerifyODataIsRegistered(app);
+
+            return app.UseMvc(b =>
+            {
+                b.Select().Expand().Filter().OrderBy().MaxTop(100).Count();
```

Comment:
nit: is there a default top variable used somewhere instead of hardcoding 100?

Reply:
It seems no.